ferrocene/doc/requirements/src/self-test.rst

```diff
@@ -0,0 +1,209 @@
+
+   - ``$target`` refers to any of the targets listed
+   - ``$hash`` is a 16 character hexadecimal string
+
+All targets
+^^^^^^^^^^^
+
+- Inside of ``lib/rustlib/$target/lib`` directory,
+  relative to the root of the installation,
+  should exist these regular files:
+
+  - ``libcore-$hash.rlib``
+  - ``liballoc-$hash.rlib``
+
+- Inside of ``lib/rustlib/$target/lib`` directory,
+  relative to the root of the installation,
+  should not exist duplicates of these regular files:
+
+  - ``libcore-$hash.rlib``
+  - ``liballoc-$hash.rlib``
+
+Targets with ``std``
+^^^^^^^^^^^^^^^^^^^^
+
+- Inside of ``lib/rustlib/$target/lib`` directory,
+  relative to the root of the installation,
+  should exist these regular files:
+
+  - ``libstd-$hash.rlib``
+  - ``libtest-$hash.rlib``
+  - ``libproc_macro-$hash.rlib``
+
+- Inside of ``lib/rustlib/$target/lib`` directory,
+  relative to the root of the installation,
+  should not exist duplicates of these regular files:
+    
+  - ``libstd-$hash.rlib``
+  - ``libtest-$hash.rlib``
+  - ``libproc_macro-$hash.rlib``
+
+Checks for linkers
+------------------
+
+.. note:: The following checks apply to these targets:
+
+   - :target:`x86_64-unknown-linux-gnu`
+
+   ``$target`` refers to any of the targets listed above
+
+Check if we can create temporary directories.
+
+Inside of ``lib/rustlib/$target/bin`` directory,
+relative to the root of the installation,
+should exist the regular file named ``rust-lld``.
+
+Inside of ``lib/rustlib/$target/bin/gcc-ld`` directory,
+relative to the root of the installation,
+should exist the regular file named ``ld.lld``,
+which is the linker wrapper.
+
+Search for a system C compiler in the ``PATH`` environment variable.
+
+Use the system C compiler to compile a sample program,
+and use the linker wrapper for the linking stage.
+
+Check that the system C compiler passes ``-Wl,$arg`` arguments to the linker,
+where ``$arg`` is command line arguments that the system linker accepts.
+
+Checks for compilation
+----------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Target
+     - Host target
+   * - :target-with-triple:`x86_64-unknown-linux-gnu`
+     - ✔
+   * - :target-with-triple:`aarch64-unknown-none`
+     -
+   * - :target-with-triple:`x86_64-pc-windows-msvc`
+     - ✔
+   * - :target-with-triple:`aarch64-unknown-linux-gnu`
+     - ✔
+   * - :target-with-triple:`aarch64-apple-darwin`
+     - ✔
+   * - :target-with-triple:`x86_64-apple-darwin`
+     -
+   * - :target-with-triple:`aarch64-unknown-nto-qnx710`
+     -
+   * - :target-with-triple:`x86_64-pc-nto-qnx710`
+     -
+   * - :target-with-triple:`armv8r-none-eabihf`
+     -
+   * - :target-with-triple:`thumbv7em-none-eabi`
+     -
+   * - :target-with-triple:`thumbv7em-none-eabihf`
+     -
+   * - :target-with-triple:`armv7r-none-eabihf`
+     -
+   * - :target-with-triple:`armebv7r-none-eabihf`
+     -
+
+All targets
+^^^^^^^^^^^
+
+Check if we can create temporary directories.
```

Comment:
I think we should keep it.

We have an error code for it: https://public-docs.ferrocene.dev/main/user-manual/self-test/error-codes.html#fls-017-creation-of-temporary-directory-failed. An I understood that "we check what is in the requirements, not more, not less". So I think we either should have it in the requirements, or not emit an error code for it. But since it is a potential failure if the user actually cannot create temp dirs I think we should keep it in the requirements.

Reply:
It's true that everything the tool is supposed to check should be a requirement. That error code though is not something we explicitly check, it's just a failure condition that happens when we create a temporary directory to try and compile programs.

Reply:
7d3bfa96634